pymatgen/core/surface.py

```diff
@@ -869,63 +917,71 @@ def __init__(
         self.min_slab_size = min_slab_size
         self.in_unit_planes = in_unit_planes
         self.primitive = primitive
-        self._normal = normal
+        self._normal = normal  # TODO (@DanielYang59): used only in unit test
```

Comment:
This protected attribute `self._normal` seems only used for unit test (to confirm surface normal is indeed perpendicular to vector a/b): https://github.com/materialsproject/pymatgen/blob/a2ab944bf8e1b90e51ca2b65f853972392177286/tests/core/test_surface.py#L400-L404 

This is certainly not wrong but I'm just wondering if there is a cleaner way to achieve this? As having some code here just for test purpose "smells suspicious", but I don't know how to do better frankly.

Reply:
normal vector seems like a useful thing to know about a surface. surprised that it's only used in testing. we could turn it into a `@property` of `Slab` and give it it's own unit test

Reply:
Thanks for sharing your thoughts. It's actually a property for `Slab` already, see: https://github.com/materialsproject/pymatgen/blob/a2ab944bf8e1b90e51ca2b65f853972392177286/pymatgen/core/surface.py#L410-L415

Reply:
i see and https://github.com/materialsproject/pymatgen/blob/a2ab944bf8e1b90e51ca2b65f853972392177286/pymatgen/core/surface.py#L872 is on `SlabGenerator`, not `Slab` itself. but shouldn't it also be a `@property` on `SlabGenerator`?

Reply:
I'm still reading and consuming the `core.surface` module, so I may not be having a good enough understanding. And I'm thinking of cleaning up this module "superficially" in the first go here and then dive deeper later, right?

The `SlabGenerator` seems like a generator (quite obviously) to generator `Slab`s from a given condition, as such I wouldn't expect the generator to carry the normal vector of the `Slab` (which sounds like a property of the `Slab` itself). What do you think?

Reply:
> I wouldn't expect the generator to carry the normal vector of the Slab

but it already does? unless you want to remove it in which case i'm open to arguments. it would be good to know if all generated slabs are guaranteed to have the same normal vector. in that case, i think it makes sense to have it on the generator. if not, i don't understand why it's there in the 1st place

Reply:
> but it already does? 

Yes it does but as a protected attribute instead of a `property`.

> it would be good to know if all generated slabs are guaranteed to have the same normal vector.

I think this should be an internal checking mechanism for `SlabGenerator`. Currently the checking mechanism for surface normal seems rely on the external unit test, which feels weird (and thus I tagged it here), maybe we should migrate this check into `SlabGenerator`?

> it would be good to know if all generated slabs are guaranteed to have the same normal vector.

I'm not certain about this, but at least from the original author and also the [original paper](https://www.sciencedirect.com/science/article/pii/S003960281300160X): `Normality is not guaranteed, but the oriented cell will have the c vector as normal as possible to the surface.` here: https://github.com/materialsproject/pymatgen/blob/a2ab944bf8e1b90e51ca2b65f853972392177286/pymatgen/core/surface.py#L759-L770

Maybe we could put it aside for now and revisit this after I finish consuming the entire `core.surface` module (By which time I would have a much better understanding of all these)? Sorry for drawing your attention too early.

Reply:
sounds good